app/bundles/LeadBundle/EventListener/ReportSubscriber.php

```diff
@@ -438,8 +438,24 @@ public function onReportGraphGenerate(ReportGraphEvent $event)
             $chartQuery->applyDateFilters($queryBuilder, 'date_added', 'l');
 
             if ('lp' === $queryBuilder->getQueryPart('from')[0]['alias']) {
+                $join = $queryBuilder->getQueryPart('join');
                 $queryBuilder->resetQueryPart('join');
+
                 $queryBuilder->leftJoin('lp', MAUTIC_TABLE_PREFIX.'leads', 'l', 'l.id = lp.lead_id');
+                if (isset($join['l'])) {
+                    foreach ($join['l'] as $item) {
+                        switch ($item['joinType']) {
+                            case 'left':
+                                $queryBuilder->leftJoin('l', $item['joinTable'], $item['joinAlias'], $item['joinCondition']);
+                                break;
+                            case 'right':
+                                $queryBuilder->rightJoin('l', $item['joinTable'], $item['joinAlias'], $item['joinCondition']);
+                                break;
+                            default:
+                                $queryBuilder->join('l', $item['joinTable'], $item['joinAlias'], $item['joinCondition']);
+                        }
+                    }
+                }
```

Comment:
Perhaps you could use this instead:
```php
return $queryBuilder->add('join', [
            'l' => [
                'joinType'      => $item['joinType'],
                'joinTable'     => $item['joinTable'],
                'joinAlias'     => $item['joinAlias'],
                'joinCondition' => $item['joinCondition'],
            ],
        ], true);
```
But it should be also added only when there is some field used with the `l.*` column to keep the optimization in.

Reply:
I just use the `join` clause now. I added an additional condition, but not `l.*`. Since `l.*` is used in the line above, I don't check anything with it.